### DISCUSSION TITLE
File and Folder Grouping View

### DISCUSSION THREAD

**Discussion by kannasama** [spacedriveapp/spacedrive#2055] [Feature Request]
Something I think just about every file manager has lacked across all platforms is the ability to sort and group files and folders by various attributes, i.e. date, type, alphabetically.  Windows Explorer has done this since Windows Vista, but I've yet to see a similar implementation elsewhere.  Would very much like to see it implemented here, as I'm impressed with what I've seen so far.

**Comment by ioqbit** [comment]
It is already possible to sort by attributes.

It’s just that currently no distinction has been made between empty folders and folders with content. You must enter the folder to know whether there is content in the folder.

**Reply by kannasama** [reply]
There's still the grouping aspect that doesn't exist yet, a la Windows.  Ideally, the sort and grouping would also be independent, for example:
- Group by type
- Sort by name in each group

Among other pairings based on the sortable attributes.

**Comment by utkubakir** [comment]
Grouping, searching and filtering can always be improved and several variations of these are already in our roadmap, thank you for the suggestion!
